integrations/csm-notification-service/internal/handler/events_test.go

```
@@ -0,0 +1,147 @@
+
+func assertStatus(t *testing.T, w *httptest.ResponseRecorder, want int) {
+	t.Helper()
+	if w.Code != want {
+		t.Errorf("status = %d, want %d (body=%s)", w.Code, want, w.Body.String())
+	}
+}
+
+func postEvent(t *testing.T, pub *mockPublisher, body string) *httptest.ResponseRecorder {
+	t.Helper()
+	h := NewEventsHandler(pub)
+	r := httptest.NewRequest(http.MethodPost, "/events", strings.NewReader(body))
```

golangci-lint (2.12.2)

[error] net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext

(noctx)